sql/src/main/java/io/druid/sql/avatica/DruidStatement.java

```diff
@@ -83,6 +94,7 @@ public DruidStatement(
     this.statementId = statementId;
     this.queryContext = queryContext == null ? ImmutableMap.of() : queryContext;
     this.onClose = Preconditions.checkNotNull(onClose, "onClose");
+    this.yielderOpenCloseExecutor = Executors.newSingleThreadExecutor();
```

Comment:
Give this a name, and use `Execs.singleThreaded(name)` which (a) makes the naming simpler, (b) makes sure the thread is a daemon thread.